planner/core/casetest/physicalplantest/physical_plan_test.go

```diff
@@ -2589,3 +2589,71 @@ func TestIndexMergeOrderPushDown(t *testing.T) {
 		tk.MustQuery("show warnings").Check(testkit.Rows(output[i].Warning...))
 	}
 }
+
+func TestExplainNonEvaledSubquery(t *testing.T) {
```

Comment:
```physicalplantest``` also has too many tests. so I suggest you create a new package to put the case test of the scalarSubQuery.